host/src/gap.rs

```diff
@@ -0,0 +1,122 @@
+    /// Smart Watch device appearance.
+    pub const SMART_WATCH: [u8; 2] = new(0x003, 0x020);
+    /// Generic Power device appearance.
+    pub const GENERIC_POWER: [u8; 2] = new(0x01E, 0x000);
+    /// Generic Sensor device appearance.
+    pub const GENERIC_SENSOR: [u8; 2] = new(0x015, 0x000);
+    /// Gamepad device appearance.
+    pub const GAMEPAD: [u8; 2] = new(0x00F, 0x040);
+}
+
+/// Configuration for the GAP Service.
+pub enum GapConfig {
+    /// Peripheral device configuration.
+    Peripheral(PeripheralConfig),
+    /// Central device configuration.
+    Central(CentralConfig),
+}
+
+/// Configuration for a peripheral device GAP Service.
+pub struct PeripheralConfig {
+    /// The name of the peripheral device.
+    pub name: &'static str,
+    /// The representation of the external appearance of the device.
+    ///
+    /// /// Example: `&appearance::GENERIC_SENSOR`
+    pub appearance: &'static [u8; 2],
+    // TODO: Add more GAP parameters
+    // pub preferred_connection_parameters: Option<ConnectionParameters>,
+}
+
+/// Configuration for a central device GAP Service.
+pub struct CentralConfig {
+    /// The name of the central device.
+    pub name: &'static str,
+    /// The representation of the external appearance of the device.
+    ///
+    /// Example: `&appearance::GENERIC_SENSOR`
+    pub appearance: &'static [u8; 2],
+    // TODO: Add more GAP parameters
+}
+
+impl GapConfig {
+    /// Create a default peripheral configuration.
+    ///
+    /// This configuration will use the `GENERIC_UNKNOWN` appearance.
+    pub fn default(name: &'static str) -> Self {
+        GapConfig::Peripheral(PeripheralConfig {
+            name,
+            appearance: &appearance::GENERIC_UNKNOWN,
+        })
+    }
+    /// Add the GAP service to the attribute table.
+    pub fn build<M: RawMutex, const MAX: usize>(self, table: &mut AttributeTable<'_, M, MAX>) {
+        // Service UUIDs.  These are mandatory services.
+        const GAP_UUID: u16 = 0x1800;
+        const GATT_UUID: u16 = 0x1801;
+
+        // Characteristic UUIDs.  These are mandatory characteristics.
+        const DEVICE_NAME_UUID: u16 = 0x2a00;
+        const APPEARANCE_UUID: u16 = 0x2a01;
+
+        let mut gap = table.add_service(Service::new(GAP_UUID)); // GAP UUID (mandatory)
+        match self {
+            GapConfig::Peripheral(config) => {
+                let id = config.name.as_bytes();
+                let _ = gap.add_characteristic_ro(DEVICE_NAME_UUID, id);
+                let _ = gap.add_characteristic_ro(APPEARANCE_UUID, &config.appearance[..]);
+            }
+            GapConfig::Central(config) => {
+                let id = config.name.as_bytes();
+                let _ = gap.add_characteristic_ro(DEVICE_NAME_UUID, id);
+                let _ = gap.add_characteristic_ro(APPEARANCE_UUID, &config.appearance[..]);
+            }
+        };
+        gap.build();
+
+        table.add_service(Service::new(GATT_UUID)); // GATT UUID (mandatory)
```

Comment:
@alexmoon is this what you mean about the Gatt service? At present its created with no characteristics, as per @lulf's pre-existing boilerplate.

Reply:
Ah, yes. Sorry, I missed it.